client/ayon_harmony/plugins/publish/validate_instances.py

```diff
@@ -41,6 +41,7 @@ class ValidateInstance(pyblish.api.InstancePlugin):
     hosts = ["harmony"]
     actions = [ValidateInstanceRepair]
     order = ValidateContentsOrder
+    optional = True
 
     def process(self, instance):
         instance_folder_path = instance.data["folderPath"]
```

Comment:
This should still do `is_active(instance.data)` using `OptionalPyblishPluginMixin`, etc.

Reply:
> this check should be held by the parent class by default

Hardly doable I think, because it'd need to be able to interrupt your call to `process()` (because we're overriding that). Only thing there would be working with metaclasses that inject the logic around `process` even if you were to override it in a subclass - but that's just more magical bits prone to failure.

It's a side effect of building on Pyblish that in itself has no notion of 'per instance - per plugin' overrides other than instance and context data inside `process()`.

Happy to let @iLLiCiTiT fight this one out further if you need.

Reply:
Long story short, we're by-passing most of the pyblish logic. So we either heavily modify pyblish and break backwards compatibility (and it would not be pyblish anymore), move the pyblish code to ayon-core, do modifications we need, and break backwards compatibility, or use this approach.